studio/schemas/documents/compensations.ts

```diff
@@ -15,24 +16,29 @@ const compensations = defineType({
   type: "document",
   title: "Compensations",
   fields: [
-    locations,
     title,
     titleSlug,
-    seo,
+    locations,
     defineField({
-      name: "showSalaryCalculator",
-      title: "Show Salary Calculator",
-      description: "Should the salary calculator be visible on the page?",
-      type: "boolean",
-      initialValue: true,
+      ...compensationDetails,
+      readOnly: ({ parent }) => !parent?.showEstimatedSalary,
```

Comment:
Why do we need to make it read only? This forces the content editor to show the estimated salary in the frontend to edit compensation details.

Reply:
Sanity isn't built on the draft principle, that I assume you want the creator to have ownership over. I don't see a situation where that's something Variant would benefit from. 

So we're setting it to read only to indicate that the fields are irrelevant if you're not interested in showing the calculator in production. 

Nothing is added to production until it is published. So if a creator for some reason would like to add content to the compensationDetails while have benefits in production. They can untoggle the show estimated salary, add content, and then toggle it back to false when ready. 

However, that's not a typical CMS type of workflow. What's more common is to create a new document, add all kinds of data and then unpublish v1 and publish v2 once you're done. Sanity does have a bulk document publishing feature in the growth plan for this.

Reply:
What about moving "show estimate toggle button" above "compnesation details", and hide this information until the show estimate salary is set to true?

Reply:
I would go for hiding instead of readonly in that case.

Reply:
I agree we can move it above! 

I wouldn1t want to hide the content based on a boolean value in this context. (There are other places I'd probably go back and visit in this repository). Because the hidden content might not be easily accessible for users relying on screen readers or other assistive technologies, I'd want them to see it as read only when disabled (assuming sanity has their a11y up to date). That way they have access to the user experience of the schema, and can easily tell whether or not the toggle should be true or false.

Reply:
Would it be clear why the fields are read-only in that case (e.g. "toggle show salary calculator to enable this field")?